modules/public/views.py

```diff
@@ -193,29 +235,44 @@ def search_snap():
 
     page = floor(offset / size) + 1
 
-    searched_results = api.get_searched_snaps(
-        quote_plus(snap_searched),
-        size,
-        page
-    )
+    errors = []
+    normalize_results = []
+    links = []
+    try:
+        searched_results = api.get_searched_snaps(
+            quote_plus(snap_searched),
+            size,
+            page
+        )
 
-    context = {
-        "query": snap_searched,
-        "snaps": normalize_searched_snaps(searched_results),
-        "links": get_pages_details(
+        normalize_results = normalize_searched_snaps(searched_results)
+        links = get_pages_details(
             (
                 searched_results['_links']
                 if '_links' in searched_results
                 else []
             )
         )
+    except api.ApiResponseError as api_error_exception:
+        if hasattr(api_error_exception, 'errors'):
+            errors = api_error_exception.errors
+        else:
+            errors.append(str(api_error_exception))
```

Comment:
Sorry if this was because of my suggestion, but we don't need `errors.append()` in this case - we're just making one call which will just return one error.